### PR TITLE
Upgrade openapi after subscriptions removal

### PR DESCRIPTION
**Pull request by jtherrmann**
See https://github.com/ASFHyP3/hyp3/issues/1193

After removing the Subscriptions feature, we've significantly changed our API spec and have removed the need for WKT string validation, which makes much of the work done on https://github.com/ASFHyP3/hyp3/pull/1708 obsolete, so I'm starting over from scratch with this PR.

This PR includes the following changes:

- Upgrade the `openapi-core`, `openapi-spec-validator`, and `jsonschema` packages to their latest versions
- Minor updates to the reference paths in our OpenAPI spec files
- Convert our API endpoints from Flask class-based views to functions decorated with `@app.route`, which leads to simpler code (in my opinion) and, more importantly, allows us to more easily use the `FlaskOpenAPIViewDecorator` decorator provided by `openapi-core`, which in turn allows us to re-implement the following features according to the latest `openapi-core` interface:
    - Disabling of response validation
    - Custom error formatting

TODO:
- [ ] Create an issue for converting the `openapi-core` pin back to the normal format following the next PyPI release (see the `TODO` comment in the requirements file)

### CODE REVIEW

**Comment by jtherrmann** [comment]
The `cfn-lint` package pins `jsonschema` [here](https://github.com/aws-cloudformation/cfn-lint/blob/main/setup.py) (see https://github.com/aws-cloudformation/cfn-lint/pull/2792).

**Comment by jtherrmann** [comment]
The version conflict should be resolved by the most recent version of `cfn-lint`.

**Comment by jtherrmann** [comment]
We can now upgrade the `openapi-*` and `jsonschema` packages to their latest versions.

File: apps/api/src/hyp3_api/api-spec/job_parameters.yml.j2 [hunk 1]
```diff
@@ -53,7 +53,7 @@ components:
           enum:
             - {{ job_type }}
         name:
-          $ref: "./openapi-spec.yml#components/schemas/name"
+          $ref: "./openapi-spec.yml#/components/schemas/name"
```
Comment (forrestfwilliams) [inline apps/api/src/hyp3_api/api-spec/job_parameters.yml.j2:56]: I see the additional of this `/` in several files. Is this an update to how our yamls work?

File: apps/api/src/hyp3_api/routes.py [hunk 2]
```diff
@@ -68,7 +67,7 @@ def render_ui():
 
 
 @app.errorhandler(404)
-def error404(e):
+def error404(_):
```
Comment (forrestfwilliams) [inline apps/api/src/hyp3_api/routes.py:70]: I'm not a fan of an `_` as an input argument name. Is this standard practice? Could we use something like `unused_arg` instead?

**Review by forrestfwilliams — COMMENTED** [review]
From what I can tell, this PR look good. I'm confident in the changes to our configuration YAMLs, the dependabot changes, the changelog, and requirements files.

However, I have not worked with Flask APIs in the past, and thus don't feel good rubber stamping the changes in `apps/api/src/hyp3_api/routes.py`. Would @jhkennedy or @asjohnston-asf  be willing to lend a hand here?

File: apps/api/src/hyp3_api/api-spec/job_parameters.yml.j2 (hunk 1 above)
Comment (jtherrmann) [reply apps/api/src/hyp3_api/api-spec/job_parameters.yml.j2:56]: I believe it's just an update to how `jsonschema` expects reference paths to be written. Just a syntax update, nothing more.

File: apps/api/src/hyp3_api/routes.py (hunk 2 above)
Comment (jtherrmann) [reply apps/api/src/hyp3_api/routes.py:70]: I believe it is standard practice, and it gets rid of PyCharm's warning about an unused parameter.

File: apps/api/src/hyp3_api/routes.py [hunk 3]
```diff
@@ -93,67 +92,57 @@ def default(self, o):
...
+    errors_handler_cls=ErrorHandler,
+)
+
 
-    def post(self):
-        return jsonify(handlers.post_jobs(request.get_json(), g.user))
+@app.route('/jobs', methods=['POST'])
+@openapi
+def jobs_post():
+    return jsonify(handlers.post_jobs(request.get_json(), g.user))
```
Comment (jhkennedy) [inline apps/api/src/hyp3_api/routes.py:117]: This is completely nitpicky but I like the methods in alphabetical order for a route

**Review by jhkennedy — APPROVED** [review]
Looks great to me! I like the class to decorator change.

File: apps/api/src/hyp3_api/routes.py (hunk 3 above)
Comment (jtherrmann) [reply apps/api/src/hyp3_api/routes.py:117]: Fair, I'm just going to merge since I'm eager to get this deployed, but will keep in mind for the future.

**Pull request merged**